src/main/java/org/gridsuite/shortcircuit/server/dto/Filter.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.gridsuite.shortcircuit.server.dto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.List;
+
+/**
+ * An object that can be used to filter data with the JPA Criteria API (via Spring Specification)
+ * @param dataType the type of data we want to filter (text, number)
+ * @param type the type of filter (contains, startsWith...)
+ * @param value the value of the filter
+ * @param column the column / field on which the filter will be applied
+ */
+public record Filter(DataType dataType, Type type, Object value, String column) {
```

Comment:
Rename to a more specific name ?
It should contain "FeederResult" ?
and we already have a concept in gridsuite called "filter", maybe avoid the word filter entirely for this very different concept ? Some ideas to find something better: Search, WhereClauses, WhereCriteria,  DomainSpecification, Selector, ...?

E.G.  FeederResultSearch ?

Reply:
It's a generic record that will be used to filter Faults too, so I can't add Feeder in it.
I can use Selector as it's already used in Sensi...